xla/service/gpu/gpu_executable.cc

```diff
@@ -343,6 +365,23 @@ absl::Status ExecuteThunks(const std::string& module_name,
     command_buffer_trace_stream = borrowed_command_buffer_trace_stream->get();
   }
 
+  // Borrow stream for additional compute streams
+  absl::flat_hash_map<ExecutionStreamId, se::Stream*> additional_compute_streams;
+  if (additional_compute_stream_ids.size() > 0) {
```

Comment:
With this branch you return borrowed streams back to the pool immediately, and can accidentaly reuse the same thread for many concurrent runs, `additional_streams` lifetime should match `additional_compute_streams`, I suggest to simply remove this check.

Reply:
ah good catch. removed the check.